util/targen/ldd.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package targen
+
+import (
+	"bufio"
+	"bytes"
+	"path/filepath"
+
+	"github.com/coreos/mantle/system/exec"
+)
+
+// return a slice of strings that are the library dependencies of binary.
+func ldd(binary string) ([]string, error) {
+	c := exec.Command("ldd", binary)
```

Comment:
This only works for native binaries because ldd is just invoking the program with special linker args.